website/static/js/registrationUtils.js

```diff
@@ -561,38 +561,36 @@ RegistrationEditor.prototype.check = function() {
                 }
             }
         });
-    }
-    else {
+    } else {
         window.location = self.draft().urls.register_page;
     }
 };
-// TODO: uncomment when we support commenting
-//RegistrationEditor.prototype.viewComments = function() {
-//    var self = this;
-//    
-//    var comments = self.currentQuestion().comments();
-//    $.each(comments, function(index, comment) {
-//        if (comment.seenBy().indexOf(currentUser.id) === -1) {
-//            comment.seenBy.push(currentUser.id);
-//        }
-//    });
-//};
-//RegistrationEditor.prototype.getUnseenComments = function(qid) {
-//    var self = this;
-//
-//    var question = self.draft().schemaData[qid];
-//    var comments = question.comments || [];
-//    for (var key in question) {
-//        if (key === 'comments') {
-//            for (var i = 0; i < question[key].length - 1; i++) {
-//                if (question[key][i].indexOf(currentUser.id) === -1) {
-//                    comments.push(question[key][i]);
-//                }
-//            }
-//        }
-//    }
-//    return comments;
-//};
+RegistrationEditor.prototype.viewComments = function() {
+    var self = this;
+
+    var comments = self.currentQuestion().comments();
+    $.each(comments, function(index, comment) {
+        if (comment.seenBy().indexOf(currentUser.id) === -1) {
+            comment.seenBy.push(currentUser.id);
+        }
+    });
+};
+RegistrationEditor.prototype.getUnseenComments = function(qid) {
+    var self = this;
+
+    var question = self.draft().schemaData[qid];
+    var comments = question.comments || [];
+    for (var key in question) {
+        if (key === 'comments') {
```

Comment:
As far as I can tell, `question` comes from a `DictField` (see https://github.com/HarryRybacki/osf.io/blob/feature/prereg-prize/website/project/model.py#L3670 for structure).

If that's correct, it would probably be more efficient to access the value of `question.comments` directly instead of iterating and comparing keys. It's possible that I'm missing something though.

See `draft().schemaData` definition: https://github.com/HarryRybacki/osf.io/blob/feature/prereg-prize/website/static/js/registrationUtils.js#L306

And view that populates the field: https://github.com/HarryRybacki/osf.io/blob/feature/prereg-prize/website/project/views/drafts.py#L166